waves-integration/src/main/scala/com/wavesplatform/dex/grpc/integration/clients/combined/CombinedStreamActor.scala

```diff
@@ -0,0 +1,281 @@
+package com.wavesplatform.dex.grpc.integration.clients.combined
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.{Behaviors, StashBuffer}
+import com.wavesplatform.dex.actors.CustomLoggerBehaviorInterceptor
+import com.wavesplatform.dex.fp.PartialFunctionOps.Implicits
+import com.wavesplatform.dex.grpc.integration.clients.ControlledStream.SystemEvent
+import com.wavesplatform.dex.grpc.integration.clients.blockchainupdates.{BlockchainUpdatesControlledStream, BlockchainUpdatesConversions}
+import com.wavesplatform.dex.grpc.integration.clients.combined.CombinedStream.{Settings, Status}
+import com.wavesplatform.dex.grpc.integration.clients.domain.WavesNodeEvent
+import com.wavesplatform.dex.grpc.integration.clients.matcherext.{UtxEventConversions, UtxEventsControlledStream}
+import com.wavesplatform.dex.grpc.integration.services.UtxEvent
+import com.wavesplatform.events.api.grpc.protobuf.SubscribeEvent
+import monix.execution.Scheduler
+import monix.reactive.subjects.ConcurrentSubject
+import org.slf4j.event.Level
+
+import java.util.concurrent.atomic.AtomicInteger
+
+object CombinedStreamActor {
+  sealed trait Message extends Product with Serializable
+
+  sealed trait Command extends Message
+
+  object Command {
+    case class Start(fromHeight: Int) extends Command
+
+    case class UpdateProcessedHeight(height: Int) extends Command
+
+    case object Restart extends Command
+
+    case object Continue extends Command
+
+    case class ProcessUtxSystemEvent(evt: SystemEvent) extends Command
+    case class ProcessBlockchainUpdatesSystemEvent(evt: SystemEvent) extends Command
+
+    case class ProcessUtxEvent(evt: UtxEvent) extends Command
+    case class ProcessBlockchainUpdatesEvent(evt: SubscribeEvent) extends Command
+  }
+
+  def apply(
+    settings: Settings,
+    processedHeight: AtomicInteger,
+    blockchainUpdates: BlockchainUpdatesControlledStream,
+    utxEvents: UtxEventsControlledStream,
+    status: ConcurrentSubject[Status, Status],
+    output: ConcurrentSubject[WavesNodeEvent, WavesNodeEvent]
+  )(implicit monixScheduler: Scheduler): Behavior[Command] = Behaviors.setup[Command] { context =>
+    // Subscribe on events
+
+    blockchainUpdates.systemStream.foreach(context.self ! Command.ProcessBlockchainUpdatesSystemEvent(_))
+    blockchainUpdates.stream.foreach(context.self ! Command.ProcessBlockchainUpdatesEvent(_))
+
+    utxEvents.systemStream.foreach(context.self ! Command.ProcessUtxSystemEvent(_))
+    utxEvents.stream.foreach(context.self ! Command.ProcessUtxEvent(_))
+
+    // Utilities
+
+    val ignore: Command => Behavior[Command] = _ => Behaviors.same
+    def partial(f: PartialFunction[Command, Behavior[Command]]): PartialFunction[Command, Behavior[Command]] = f
```

Comment:
It looks like this, but it doesn't because it helps the compiler to [determine types](https://scastie.scala-lang.org/u26RmGwSRj23TR3tAM5cXQ). I added the `PartialFunctionOps.mkPartial` function outside the implicit.